pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMessageDecoder.java

```diff
@@ -46,8 +47,22 @@ public interface StreamMessageDecoder<T> {
    * @param topicName Topic name of the stream
    * @throws Exception If an error occurs
    */
-  void init(Map<String, String> props, Set<String> fieldsToRead, String topicName)
-      throws Exception;
+  default void init(Map<String, String> props, Set<String> fieldsToRead, String topicName)
+      throws Exception {
+    throw new UnsupportedOperationException("init method not implemented");
+  }
+
+  /**
+   * Initializes the decoder.
+   * @param streamConfig Can be derived from tableConfig but is passed explicitly to avoid redundant computation
+   * @param tableConfig Table Config of the table
+   * @param schema Schema of the table
+   * @throws Exception
+   */
+  default void init(@Nullable Set<String> fields, StreamConfig streamConfig, TableConfig tableConfig, Schema schema)
```

Comment:
Just realize `fields` won't be `null`
```suggestion
  default void init(Set<String> fieldsToRead, StreamConfig streamConfig, TableConfig tableConfig, Schema schema)
```